src/OpenLoco/Industry.cpp

```diff
@@ -240,12 +244,119 @@ namespace OpenLoco
                     }
                 }
 
-                const auto remainingCargo = StationManager::deliverCargoToStations(stations, indObj->produced_cargo_type[i], max);
-                var_19F[i] = Math::Bound::add(remainingCargo, var_19F[i]);
+                const auto quantityDelivered = StationManager::deliverCargoToStations(stations, indObj->produced_cargo_type[i], max);
+                producedCargoQuantityDeliveredMonthlyTotal[i] = Math::Bound::add(quantityDelivered, producedCargoQuantityDeliveredMonthlyTotal[i]);
             }
         }
     }
 
+    // 0x00453868
+    void Industry::updateMonthly()
+    {
+        if (flags & IndustryFlags::flag_01)
+        {
+            return;
+        }
+
+        if (flags & IndustryFlags::closingDown && var_17D[0] == 0 && var_17D[1] == 0)
+        {
+            GameCommands::do_48(GameCommands::Flags::apply, id());
+            return;
+        }
+        bool hasEvent = false;
+        const auto* indObj = getObject();
+        if (under_construction == 0xFF
+            && !(flags & IndustryFlags::closingDown)
+            && indObj->required_cargo_type[0] == 0xFF)
+        {
+            if (indObj->flags & IndustryObjectFlags::unk18
+                && producedCargoPercentTransportedPreviousMonth[0] > 70
+                && gPrng().randNext(31) == 0
+                && productionRate[0] < 100
+                && productionRate[1] < 100)
+            {
+                productionRate[0] = std::min(100, productionRate[0] * 2);
+                productionRate[1] = std::min(100, productionRate[1] * 2);
+                MessageManager::post(MessageType::industryProductionUp, CompanyId::null, enumValue(id()), 0xFFFF);
+                hasEvent = true;
+            }
+            else if (indObj->flags & IndustryObjectFlags::unk19 && ((producedCargoPercentTransportedPreviousMonth[0] > 50 && productionRate[0] > 20 && gPrng().randNext(31) == 0) || (producedCargoPercentTransportedPreviousMonth[0] <= 50 && productionRate[0] > 10 && gPrng().randNext(15) == 0)))
```

Comment:
Could we split this into several lines? It's a little wide right now.

Reply:
I've split them all off into separate functions.